homeassistant/components/template/button.py

```diff
@@ -121,11 +120,8 @@ def __init__(
         """Initialize the button."""
         super().__init__(hass, config=config, unique_id=unique_id)
         assert self._attr_name is not None
-        self._command_press = (
-            Script(hass, config.get(CONF_PRESS), self._attr_name, DOMAIN)
-            if config.get(CONF_PRESS, None) is not None
-            else None
-        )
+        if (action := config.get(CONF_PRESS)) is not None:
```

Comment:
The `_action_scripts` dict can only have Script objects, the specific None check is not needed. Same comment for other classes.